fdbserver/storageserver.actor.cpp

```diff
@@ -2209,10 +2224,17 @@ ACTOR Future<Version> watchWaitForValueChange(StorageServer* data, SpanContext p
 
 	state Version minVersion = data->data().latestVersion;
 	state Future<Void> watchFuture = data->watches.onChange(metadata->key);
+	state Future<Void> tenantWatchFuture;
+	if (tenantId != TenantInfo::INVALID_TENANT) {
+		tenantWatchFuture = data->tenantWatches.onChange(tenantId);
+	}
 	state ReadOptions options;
 	loop {
 		try {
-			metadata = data->getWatchMetadata(key);
+			if (tenantId != TenantInfo::INVALID_TENANT) {
+				data->checkTenantEntry(latestVersion, TenantInfo(tenantId, Optional<Standalone<StringRef>>()));
```

Comment:
On the first iteration throwing `tenant_not_found` is reasonable, but on subsequent iterations I think we want to throw `tenant_removed`.